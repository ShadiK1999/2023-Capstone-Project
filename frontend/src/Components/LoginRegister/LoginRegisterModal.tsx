import { Modal } from '..';
import { RegistrationPanel, LoginPanel } from '.';
import { FC, useMemo } from 'react';

export type View = 'LOGIN' | 'REGISTER';

type ComponentProps = {
  view: View;
  setView: (view: View) => void;
  isOpen: boolean;
  onClose: () => void;
};

export const LoginRegisterModal: FC<ComponentProps> = ({ view, setView, isOpen, onClose }) => {
  const viewProperties: Record<
    View,
    {
      page: JSX.Element;
      title: string;
      message: string;
      nextViewbuttonText: string;
      nextView: View;
    }
  > = useMemo(
    () => ({
      LOGIN: {
        page: <LoginPanel onClose={onClose} />,
        title: 'Login',
        message: 'Register for an account? ',
        nextViewbuttonText: 'Register',
        nextView: 'REGISTER',
      },
      REGISTER: {
        page: <RegistrationPanel onClose={onClose} />,
        title: 'Register',
        message: 'Already have an account? ',
        nextViewbuttonText: 'Login',
        nextView: 'LOGIN',
      },
    }),
    [onClose],
  );

  const { page, title, message, nextViewbuttonText, nextView } = viewProperties[view];

  return (
    <Modal title={title} isOpen={isOpen} onClose={onClose}>
      {page}
      <br />
      <label>
        {message}
        <a onClick={() => setView(nextView)} href="#">
          {nextViewbuttonText}
        </a>
      </label>
    </Modal>
  );
};
