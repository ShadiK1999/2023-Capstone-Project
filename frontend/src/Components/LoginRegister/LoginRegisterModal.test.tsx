import { render } from '@testing-library/react';
import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
import { LoginRegisterModal } from './LoginRegisterModal';
import { MemoryRouter } from 'react-router-dom';

vi.mock('../../App', () => ({
  router: () => {},
}));

describe('Matches DOM Snapshot', () => {
  beforeEach(() => {
    const modalRoot = document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    document.body.appendChild(modalRoot);
  });

  afterEach(() => {
    const modalRoot = document.getElementById('modal-root');
    if (modalRoot) {
      document.body.removeChild(modalRoot);
    }
  });

  it('Matches DOM Snapshot', () => {
    const tree = render(
      <MemoryRouter>
        <LoginRegisterModal
          view={'LOGIN'}
          setView={() => undefined}
          isOpen={true}
          onClose={() => undefined}
        />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Matches DOM Snapshot', () => {
    const tree = render(
      <MemoryRouter>
        <LoginRegisterModal
          view={'REGISTER'}
          setView={() => undefined}
          isOpen={true}
          onClose={() => undefined}
        />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
