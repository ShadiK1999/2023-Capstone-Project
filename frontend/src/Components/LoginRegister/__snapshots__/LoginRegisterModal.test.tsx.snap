// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches DOM Snapshot > Matches DOM Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="modal-root"
    >
      <div
        class="f1atpta9"
      >
        <h1>
          Login
        </h1>
        <div
          class="f1ur6vce"
        >
          <form>
            <div
              class="fidbxqg"
            >
              <input
                name="email"
                placeholder="Email"
                type="text"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <input
                name="password"
                placeholder="Password"
                type="password"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <label>
                <input
                  role="checkbox"
                  type="checkbox"
                />
                Remember me
              </label>
            </div>
            <div
              class="fidbxqg"
            >
              <button
                type="submit"
              >
                Login
              </button>
            </div>
          </form>
        </div>
        <br />
        <label>
          Register for an account? 
          <a
            href="#"
          >
            Register
          </a>
        </label>
        <a
          href="#"
        >
          Close
        </a>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Matches DOM Snapshot > Matches DOM Snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="modal-root"
    >
      <div
        class="f1atpta9"
      >
        <h1>
          Register
        </h1>
        <div
          class="f1ur6vce"
        >
          <form>
            <div
              class="fidbxqg"
            >
              <input
                name="name"
                placeholder="Full name"
                type="text"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <input
                name="phoneNumber"
                placeholder="Phone number"
                type="text"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <input
                name="email"
                placeholder="Email"
                type="text"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <input
                name="password"
                placeholder="Password"
                type="password"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <input
                name="confirmPassword"
                placeholder="Confirm Password"
                type="password"
              />
            </div>
            <div
              class="fidbxqg"
            >
              <label>
                <input
                  name="terms"
                  role="checkbox"
                  type="checkbox"
                />
                I agree to the terms and conditions
              </label>
            </div>
            <div
              class="fidbxqg"
            >
              <button
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
        <br />
        <label>
          Already have an account? 
          <a
            href="#"
          >
            Login
          </a>
        </label>
        <a
          href="#"
        >
          Close
        </a>
      </div>
    </div>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
